libraries/aws/provisioning/include/types/aws_iot_provisioning_types.h

```diff
@@ -316,6 +351,41 @@ typedef struct AwsIotProvisioningCreateKeysAndCertificateResponse
     } u; /**< @brief Valid member depends on operation status. */
 } AwsIotProvisioningCreateKeysAndCertificateResponse_t;
 
+/**
+ * @ingroup provisioning_datatypes_paramstructs
+ * @brief User-specific callback information for handling server response of the Provisioning CreateCertificateFromCsr
+ * service API.
+ *
+ * @paramfor @ref provisioning_function_registerthing
+ *
+ * Provides a function that is invoked on completion of an @ref AwsIotProvisioning_CreateCertificateFromCsr API
+ * operation.
+ *
+ * @initializer{AwsIotProvisioningCreateCertificateFromCsrCallbackInfo_t,AWS_IOT_PROVISIONING_CREATE_CERTIFICATE_FROM_CSR_CALLBACK_INFO_INITIALIZER}
+ */
+typedef struct AwsIotProvisioningCreateCertificateFromCsrCallbackInfo
+{
+    void * userParam; /**< The user-provided parameter that is (as the first parameter) to the callback
+                       * function (optional). */
+
+    /**
+     * @brief User-provided callback function signature.
+     *
+     * @param[in] void* #AwsIotProvisioningCreateCertificateFromCsrCallbackInfo_t.userParam
+     * @param[in] AwsIotProvisioningCallbackParam_t * Parsed server response of either device credentials
+     * or provisioned device information.
+     *
+     * @see #AwsIotProvisioningCreateCertificateFromCsrResponse_t for more information on the second parameter.
+     */
+    void ( * function )( void *,
```

Comment:
We specify parameters, not just types.

Reply:
Updated all callback types to have parameter names